org.eclipse.scout.rt.rest.jersey.test/src/test/java/org/eclipse/scout/rt/rest/jersey/client/ChunkedDataTest.java

```diff
@@ -0,0 +1,198 @@
+    Response response = m_target
+        .path("string-jersey")
+        .request().get();
+
+    ChunkedInput<String> chunkedInput = response.readEntity(new GenericType<>() {
+    });
+    ChunkParser p = ChunkedInput.createParser("\n\n");
+    chunkedInput.setParser(p);
+    assertFalse(chunkedInput.isClosed());
+
+    String chunk;
+    List<String> chunks = new ArrayList<>();
+    while ((chunk = chunkedInput.read()) != null) {
+      chunks.add(chunk);
+      LOG.info("Received chunk {}", chunk);
+    }
+    assertEquals(List.of("1", "2", "3"), chunks);
+    assertTrue(chunkedInput.isClosed());
+  }
+
+  @Test
+  public void testGetDataObjectsScout() {
+    Response response = m_target
+        .path("dataobject-scout")
+        .request().get();
+
+    IChunkedDataReader<FixtureDo> reader = IChunkedDataReader.create(response, FixtureDo.class, "\n\n");
+    assertFalse(reader.isClosed());
+
+    FixtureDo chunk;
+    List<FixtureDo> chunks = new ArrayList<>();
+    while ((chunk = reader.read()) != null) {
+      chunks.add(chunk);
+      LOG.info("Received chunk {}", chunk);
+    }
+    assertChunks(chunks);
+    assertTrue(reader.isClosed());
+  }
+
+  @Test
+  public void testGetDataobjectsJersey() {
+    Response response = m_target
+        .path("dataobject-jersey")
+        .request().get();
+
+    ChunkedInput<FixtureDo> chunkedInput = response.readEntity(new GenericType<>() {
+    });
+    ChunkParser p = ChunkedInput.createParser("\n\n");
+    chunkedInput.setParser(p);
+    assertFalse(chunkedInput.isClosed());
+
+    FixtureDo chunk;
+    List<FixtureDo> chunks = new ArrayList<>();
+    while ((chunk = chunkedInput.read()) != null) {
+      chunks.add(chunk);
+      LOG.info("Received chunk {}", chunk);
+    }
+    assertChunks(chunks);
+    assertTrue(chunkedInput.isClosed());
+  }
+
+  @Test
+  public void testGetDataObjectsScout_withHeader() {
+    Response response = m_target
+        .path("dataobject-scout/header")
+        .queryParam("headerValue", "foo")
+        .request().get();
+
+    assertEquals("foo", response.getHeaderString("X-Mock"));
+
+    IChunkedDataReader<FixtureDo> reader = IChunkedDataReader.create(response, FixtureDo.class, "\n\n");
+    assertFalse(reader.isClosed());
+
+    FixtureDo chunk;
+    List<FixtureDo> chunks = new ArrayList<>();
+    while ((chunk = reader.read()) != null) {
+      chunks.add(chunk);
+      LOG.info("Received chunk {}", chunk);
+    }
+    assertChunks(chunks);
+    assertTrue(reader.isClosed());
+  }
+
+  @Test
+  public void testGetDataObjectsScout_earlyAbort() throws IOException {
+    Response response = m_target
+        .path("dataobject-scout")
+        .request().get();
+
+    IChunkedDataReader<FixtureDo> reader = IChunkedDataReader.create(response, FixtureDo.class, "\n\n");
+    assertFalse(reader.isClosed());
+
+    FixtureDo chunk = reader.read();
+    reader.close();
+    assertEquals(BEANS.get(ChunkedDataResource.class).createFixtureDo(1), chunk);
```

Comment:
It is probably difficult/impossible to test here that also the server-side (the ChunkedDataResource) stops sending?

Reply:
I dont see how to build a testcase for this behavior. Server stops sending means, no more data is written to writer.
- if the writer is not closed by the server, the client will keep on waiting for more chunks until timeout
- if the writer is closed by the server, the client will just receive the already written chunks without any other error